feature/preview/src/main/java/com/google/jetpackcamera/feature/preview/PreviewViewModel.kt

```diff
@@ -540,6 +543,28 @@ class PreviewViewModel @AssistedInject constructor(
         }
         Log.d(TAG, "captureImageWithUri")
         viewModelScope.launch {
+            val (uriIndex: Int, finalImageUri: Uri?) =
+                if (previewUiState.value is PreviewUiState.Ready &&
+                    (previewUiState.value as PreviewUiState.Ready).previewMode
+                        is PreviewMode.ExternalMultipleImageCaptureMode
+                ) {
+                    val uri =
+                        if (ignoreUri || (
+                                (previewUiState.value as PreviewUiState.Ready).previewMode
+                                    as PreviewMode.ExternalMultipleImageCaptureMode
+                                ).imageCaptureUris.isNullOrEmpty()
+                        ) {
+                            null
+                        } else {
+                            (
+                                (previewUiState.value as PreviewUiState.Ready).previewMode
+                                    as PreviewMode.ExternalMultipleImageCaptureMode
+                                ).imageCaptureUris!![externalUriIndex]
+                        }
+                    Pair(externalUriIndex, uri)
+                } else {
+                    Pair(-1, imageCaptureUri)
+                }
```

Comment:
This can be rewritten more concisely as:
```
                ((previewUiState.value as? PreviewUiState.Ready)?.previewMode as?
                        PreviewMode.ExternalMultipleImageCaptureMode)?.let {
                    val uri = if (ignoreUri || it.imageCaptureUris.isNullOrEmpty()) {
                        null
                    } else {
                        it.imageCaptureUris[externalUriIndex]
                    }
                    Pair(externalUriIndex, uri)
                } ?: Pair(-1, imageCaptureUri)
```